### ISSUE TITLE
Can't parse image references for gcr.io/cloud-builders/go:debian

### ISSUE THREAD

**Issue by bobcatfish** [tektoncd/pipeline#573]
# Expected Behavior

If I try to use a Task step that uses the image `gcr.io/cloud-builders/go:debian` without specifying a `command` explicitly, the controller should be able to fetch that image and determine what the command is (this is required for our [entrypoint rewriting logic](https://github.com/knative/build-pipeline/blob/master/docs/container-contract.md#entrypoint)).

# Actual Behavior

The controller fails to retrieve info about the image with this error:

```yaml
    - lastTransitionTime: "2019-03-01T18:08:37Z"
      message: 'References a Task default/publish-tekton-pipelines that doesn''t exist:  could
        not get entrypoint from registry for gcr.io/cloud-builders/go:debian: failed
        to fetch remote image gcr.io/cloud-builders/go:debian@sha256:bdc4c18aa7df622be964d1905c5ef60eac648c307105d23ee4b7dbbad5fe3e7d:
        couldn''t parse image gcr.io/cloud-builders/go:debian@sha256:bdc4c18aa7df622be964d1905c5ef60eac648c307105d23ee4b7dbbad5fe3e7d:
        could not parse reference'
      reason: CouldntGetTask

```

# Steps to Reproduce the Problem

1. Create a Task with a step that tries to use `gcr.io/cloud-builders/go:debian` without specifying a command, e.g. this example which is based on [go-containerregistry's cloudbuild.yaml](https://github.com/google/go-containerregistry/blob/master/cloudbuild.yaml) (it tries to install `ko`):

```yaml
  steps:
  - name: install-ko
    image: gcr.io/cloud-builders/go:debian
    env:
    - name: PROJECT_ROOT
      value: github.com/google/go-containerregistry
    args:
    - install
    - github.com/google/go-containerregistry/cmd/ko
```

2. Create a TaskRun that uses the above

# Additional Info
Note that in #564 @pivotal-nader-ziada is updating this logic anyway, maybe it will fix the issue? :sweat_smile:

**Comment by pivotal-nader-ziada** [comment]
@bobcatfish I think this is now fixed with the changes in the containers PR #564 

I get the following error from the step mentioned above
```
$kubectl logs test-task-run-pod-fe6d5a build-step-install-ko
Creating shadow workspace and symlinking source into "./gopath/src/github.com/google/go-containerregistry".
Documentation at https://github.com/GoogleCloudPlatform/cloud-builders/blob/master/go/README.md
Binaries built using 'go install' will go to "/workspace/gopath/bin".
Running: go install github.com/google/go-containerregistry/cmd/ko
can't load package: package github.com/google/go-containerregistry/cmd/ko: cannot find package "github.com/google/go-containerregistry/cmd/ko" in any of:
	/usr/local/go/src/github.com/google/go-containerregistry/cmd/ko (from $GOROOT)
	/workspace/gopath/src/github.com/google/go-containerregistry/cmd/ko (from $GOPATH)
```

which I believe means it got the command which is the problem you are having here, still doesn't work but the task needs to put the `ko` repo in the gopath

**Comment by ImJasonH** [comment]
go-containerregistry only recently gained the ability to parse references that specify both a tag and a digest (https://github.com/google/go-containerregistry/pull/391)

When we rewrite the entrypoint we just add the digest to the end of the given image, which, if it specified a tag, go-containerregistry wouldn't be able to parse.

I think updating the go-containerregistry dep could fix this.

**Comment by dlorenc** [comment]
This should have been fixed with #592.
